### PR TITLE
context-panel and systelab-date-range-picker

### PR DESCRIPTION
**Pull request by olgapuig**
Changes:
- Context-panel component. Refactor to have an AbstractContextComponent and then, an extension for the Context Menu an another for the Context Panel
- Systelab-date-range-picker to select a date range in-line
- Add filter icon to the icomoon

Use Case in HemoHub:
- Improve the usability to filter with a context panel instead of a modal dialog when there's not enough space in the screen to add all the filters.

#123

### CODE REVIEW

File: src/app/systelab-components/datepicker/datepicker.component.ts [hunk 1]
```diff
@@ -198,6 +203,7 @@ export class Datepicker implements OnInit, AfterViewInit, DoCheck, OnDestroy {
 	}
 
 	public onKeyDown(event: KeyboardEvent) {
+		event.stopPropagation();
```
Comment (alfonsserra) [inline src/app/systelab-components/datepicker/datepicker.component.ts]: Do we really need the events stop propagation?

File: src/app/systelab-components/datepicker/datepicker.component.ts [hunk 2]
```diff
@@ -214,45 +220,54 @@ export class Datepicker implements OnInit, AfterViewInit, DoCheck, OnDestroy {
...
+			if (inputElementTop + inputElementHeight + datepickerElementHeight > window.innerHeight) {
+				const newTop: number = inputElementTop + inputElementHeight - (datepickerElementHeight + inputElementHeight + 10);
+				this.myRenderer.setAttribute(element.nativeElement, 'top', newTop + 'px');
+			}
+		} catch (ex) {
 		}
 	}
 
 	public nextMonth(event: Event): void {
+		event.stopPropagation();
```
Comment (alfonsserra) [inline src/app/systelab-components/datepicker/datepicker.component.ts]: Do we really need the events stop propagation?

File: src/app/systelab-components/datepicker/datepicker.component.ts [hunk 3]
```diff
@@ -214,45 +220,54 @@ export class Datepicker implements OnInit, AfterViewInit, DoCheck, OnDestroy {
...
 		if (this.currentCalendar) {
 			this.currentCalendar.nextMonth(event);
 		}
 	}
 
 	public prevMonth(event: Event): void {
+		event.stopPropagation();
```
Comment (alfonsserra) [inline src/app/systelab-components/datepicker/datepicker.component.ts]: Do we really need the events stop propagation?

File: src/app/systelab-components/datepicker/datepicker.component.ts [hunk 4]
```diff
@@ -214,45 +220,54 @@ export class Datepicker implements OnInit, AfterViewInit, DoCheck, OnDestroy {
...
 		if (this.currentCalendar) {
 			this.currentCalendar.prevMonth(event);
 		}
 	}
 
-	public nextYear(): void {
+	public nextYear(event: Event): void {
+		event.stopPropagation();
```
Comment (alfonsserra) [inline src/app/systelab-components/datepicker/datepicker.component.ts]: Do we really need the events stop propagation?

File: src/app/systelab-components/datepicker/datepicker.component.ts [hunk 5]
```diff
@@ -214,45 +220,54 @@ export class Datepicker implements OnInit, AfterViewInit, DoCheck, OnDestroy {
...
 		if (this.currentCalendar) {
 			const currentYear = this.currentCalendar.currentYear + 1;
 			this.currentCalendar.onYearDropdownChange(currentYear.toString());
 		}
 	}
 
-	public prevYear(): void {
+	public prevYear(event: Event): void {
+		event.stopPropagation();
```
Comment (alfonsserra) [inline src/app/systelab-components/datepicker/datepicker.component.ts]: Do we really need the events stop propagation?

File: src/app/systelab-components/datepicker/datepicker.component.ts [hunk 6]
```diff
@@ -214,45 +220,54 @@ export class Datepicker implements OnInit, AfterViewInit, DoCheck, OnDestroy {
...
 		if (this.currentCalendar) {
 			const currentYear = this.currentCalendar.currentYear - 1;
 			this.currentCalendar.onYearDropdownChange(currentYear.toString());
 		}
 	}
 
-	public clearDate(event): void {
+	public clearDate(event: Event): void {
+		event.stopPropagation();
```
Comment (alfonsserra) [inline src/app/systelab-components/datepicker/datepicker.component.ts]: Do we really need the events stop propagation?

File: src/app/systelab-components/datepicker/datepicker.component.ts [hunk 7]
```diff
@@ -261,7 +276,8 @@ export class Datepicker implements OnInit, AfterViewInit, DoCheck, OnDestroy {
 		}
 	}
 
-	public setTodayDate(event): void {
+	public setTodayDate(event: Event): void {
+		event.stopPropagation();
```
Comment (alfonsserra) [inline src/app/systelab-components/datepicker/datepicker.component.ts]: Do we really need the events stop propagation?

**Comment by olgapuig** [comment]
Done! Given an inline date-picker is being used over the context-panel, the "event stop propagation" is no longer required.

**Pull request merged**